TF_Settings_Web/src/Components/Pages/Camera/Masking/displayLensFeeds.ts

```diff
@@ -27,44 +29,97 @@ export const displayLensFeeds = (
+    buf8: Uint8ClampedArray,
+    width: number,
+    lensHeight: number
+) => {
+    lensRef.width = width;
+    lensRef.height = lensHeight;
+    context.putImageData(new ImageData(buf8, width, lensHeight), 0, 0);
+};
 
-        for (let i = 0; i < width * lensHeight; i++) {
-            rightBuf32[i] = byteConversionArray[offsetView.getUint8(i)];
+export const processRotatedScreen = (
+    data: ArrayBuffer,
+    rightBuf32: Uint32Array,
+    leftBuf32: Uint32Array,
+    byteConversionArray: Uint32Array,
+    offset: number,
+    width: number,
+    lensHeight: number
+) => {
+    let rowBase = 0;
+    const offsetView = new Uint8Array(data, offset, width * lensHeight * 2);
+
+    for (let rowIndex = 0; rowIndex < width; rowIndex++) {
+        let rowStart = rowBase * 2;
+        for (let i = 0; i < lensHeight; i++) {
+            rightBuf32[i + rowBase] = byteConversionArray[offsetView[i + rowStart]];
         }
 
-        offsetView = new DataView(data.buffer.slice(offset + width * lensHeight, offset + width * lensHeight * 2));
-        for (let i = 0; i < width * lensHeight; i++) {
-            leftBuf32[i] = byteConversionArray[offsetView.getUint8(i)];
+        rowStart += lensHeight;
+        for (let i = 0; i < lensHeight; i++) {
+            leftBuf32[i + rowBase] = byteConversionArray[offsetView[i + rowStart]];
         }
+
+        rowBase += lensHeight;
     }
-    // Set black pixels to remove flashing camera bytes
-    const startOffset = isCameraReversed ? 0 : (lensHeight - 1) * width;
-    rightBuf32.fill(0xff000000, startOffset, startOffset + width);
-    leftBuf32.fill(0xff000000, startOffset, startOffset + width);
+};
 
-    leftLensRef.width = width;
-    leftLensRef.height = lensHeight;
-    leftContext.putImageData(new ImageData(leftBuf8, width, lensHeight), 0, 0);
+export const processScreen = (
+    data: ArrayBuffer,
+    rightBuf32: Uint32Array,
+    leftBuf32: Uint32Array,
+    byteConversionArray: Uint32Array,
+    offset: number,
+    width: number,
+    lensHeight: number
+) => {
+    const offsetViewRight = new Uint8Array(data, offset, width * lensHeight);
+    const offsetViewLeft = new Uint8Array(data, offset + width * lensHeight, width * lensHeight);
+    for (let i = 0; i < width * lensHeight; i++) {
+        rightBuf32[i] = byteConversionArray[offsetViewRight[i]];
+        leftBuf32[i] = byteConversionArray[offsetViewLeft[i]];
+    }
+};
 
-    rightLensRef.width = width;
-    rightLensRef.height = lensHeight;
-    rightContext.putImageData(new ImageData(rightBuf8, width, lensHeight), 0, 0);
+export const getContext = (canvasElement: HTMLCanvasElement | null) => {
+    if (!canvasElement) {
+        return undefined;
+    }
+    const canvasIndex = canvasElements.indexOf(canvasElement);
+    if (canvasIndex < 0) {
+        const context = canvasElement.getContext('2d');
+        if (context == null) {
+            return context;
+        }
+        const newIndex = canvasElements.length;
```

Comment:
React has caching inbuilt using memoization (React.memo) to avoid having to implement your own (would need to check that this works with canvasElement as it may be too complex of an input? (but probably not))